### ISSUE TITLE
[Feature] Add CentOS 8 support in rspamd.repo

### ISSUE THREAD

**Issue by zlobniyshurik** [rspamd/rspamd#3060]
## Summary
Please, add CentOS8 support in rspamd.repo

## Motivation

CentOS 8 is released. Admins wants new mail-servers with fresh soft and rspamd:)

**Comment by gralex54** [comment]
Hello,

I'm joining the request as well.

**Comment by wengler** [comment]
Hello,

Yes, Please!

**Comment by vstakhov** [comment]
That will not happen unless CentOS project has images to download. Now there are only ISO, no raw images for the containers. Dealing with ISOs is too much hassle for me, so I will wait for container images to be available.

**Comment by systemcrash** [comment]
Hmm. Understandable. The differences are possibly complex :(  Using the 7 repo, I get this:

```
# dnf --enablerepo=* install rspamd
...
Error:
 Problem: conflicting requests
  - nothing provides libevent-2.0.so.5()(64bit) needed by rspamd-1.9.4-3.x86_64
  - nothing provides libicudata.so.50()(64bit) needed by rspamd-1.9.4-3.x86_64
  - nothing provides libicui18n.so.50()(64bit) needed by rspamd-1.9.4-3.x86_64
  - nothing provides libicuuc.so.50()(64bit) needed by rspamd-1.9.4-3.x86_64
```

but
```
dnf --enablerepo=* install rspamd libevent libicu libicui18n libicuuc
...
Package libevent-2.1.8-5.el8.x86_64 is already installed.
Package libicu-60.2-7.el8.x86_64 is already installed.
No match for argument: libicui18n
No match for argument: libicuuc
Error: Unable to find a match
```

So it may be possible to install equivalent libraries and symlink to them. I've gotten a number of other CentOS 7 applications happily running on 8. An 8 package can satisfactorily be built on a 7 system (although it's less than ideal with the manual steps and symlinks probably required, since you're likely aiming for docker-type reproducability).

**Comment by systemcrash** [comment]
CentOS 8 image is live ( https://hub.docker.com/_/centos/ )

**Comment by vstakhov** [comment]
Well, I have tested it and it sucks. Many packages that are essential for Rspamd build are missing, e.g. Ragel. So I think that CentOS 8 will not be supported unless it comes to some mature state.

**Comment by bruman** [comment]
where is the latest spec file for rspamd-1.9.4-3.x86_64.rpm stored? I see the sample spec file but the version in there is 1.1.0 (https://github.com/rspamd/rspamd/blob/rspamd-1.9/centos/rspamd.spec)  so wanted to make sure that is the base for builds you are doing. 

Does ragel 7.x now work, the build instructions say to use 6.x but https://github.com/LorbusChris/rspamd-rpm/pull/6 suggests that 7.x now works. 

I am getting closer, I installed ragel from source as mentioned in the build instructions. 
Found file-devel in the PowerTools repo, 

sudo yum config-manager --set-enabled PowerTools
sudo yum install epel-release

Added some || 0%{?el8} entries to spec file 
removed dependency, on ragel  and on gmime-devel from the spec file. 
but it blows up at Directory not found rspamd/lua

I also tried https://github.com/LorbusChris/rspamd-rpm. I was able to tweak it to remove HYPERSCAN and JEMALLOC and it did successfully create an rpm, but i haven't done any testing with it.

**Comment by vstakhov** [comment]
I strongly discourage you from building your own packages as it is clear to me that you don't know what to do to make them properly. 

The recommended option is to use CentOS 7 for now, CentOS will be supported once it comes to maturity. Now it is more like piece of crap with lack of tools almost for everything.

**Comment by Wanabo** [comment]
+1 Please, add CentOS8 support in rspamd.repo

**Comment by bruman** [comment]
Let me rephrase this. 

Where is the spec file for rspamd-1.9.4-3.x86_64.rpm stored? 
Where is the associated srpm for rspamd-1.9.4-3.x86_64 ?

Does ragel 7.x now work, the build instructions say to use 6.x but a comment in LorbusChris/rspamd-rpm suggests that 7.x now works.

**Comment by vstakhov** [comment]
CentOS 8 repo is added for Rspamd experimental builds. There are no plans to add those for 1.9.4. In Rspamd 2.0, I plan to drop CentOS 6 support, Ubuntu Trusty support and add CentOS 8 support out of the box.

**Comment by gralex54** [comment]
Thanks, I've already installed it and it seems to work fine. Good job.

**Comment by rmiddle** [comment]
Were are the Experimental builds stored?

**Comment by fenice2** [comment]
> Were are the Experimental builds stored?

Here:

[rspamd-experimental]
name=Rspamd experimental repository
baseurl=http://rspamd.com/rpm/centos-8/x86_64/
enabled=1
gpgcheck=1
repo_gpgcheck=1
gpgkey=http://rspamd.com/rpm/gpg.key
